sequence_task_1/student/include/impl.h

```diff
@@ -20,31 +20,82 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    using value_type = typename Container::value_type;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using difference_type = typename Container::difference_type;
+    using size_type = typename Container::size_type;
 
     /** @todo Iterators */
+    const_iterator begin(){
+        return m_tracklist.cbegin();
+    }
+    const_iterator end(){
+        return m_tracklist.cend();
+    }
 
     /** @todo Constructor from any reversible sequence container */
+    template<class T>
+    StaticPlaylist(const T& tracklist){
+        m_tracklist.assign(tracklist.begin(), tracklist.end());
+    }
 
     /** @todo Assignment from any reversible sequence container */
+    template<class T>
+    StaticPlaylist& operator=(const T& other){
+        m_tracklist.clear();
+        m_tracklist.assign(other.begin(), other.end());
+        return *this;
+    }
 
     /** @todo Add track from initializer */
+    template<class T,class... Args>
+    const Song& play(T& first, Args&&... songData){
```

Comment:
You don't need need all these functions, if you already have the initialization data, that can be used for in-place construction.